packages/start/node/fetch.js

```diff
@@ -137,8 +137,29 @@ class NodeRequest extends BaseNodeRequest {
   }
 }
 
+/**
+ *
+ * @param {import('http').IncomingMessage} req
+ */
 export function createRequest(req) {
-  let origin = req.headers.origin || `http://${req.headers.host}`;
+  let origin;
+  if (req.httpVersionMajor === 1) {
+    origin = req.headers.origin || `http://${req.headers.host}`;
+    // It's impossible for `host` to be empty since :authority doesn't exist on HTTP/1.
+    // https://www.rfc-editor.org/rfc/rfc9110.html#section-7.2 A user agent MUST generate a Host header field in a request unless it sends that information as an ":authority" pseudo-header field
+    // We initially try the origin header because it has the scheme. The host header doesn't, and so defaults to HTTP
```

Comment:
I would ideally like to use only the host header but I don't know how to get the scheme. `createRequest` has 3 callers, but I only know how to pass in the scheme to [`packages/start/dev/server.js`](https://github.com/solidjs/solid-start/blob/8ef83015d5e1af8acc3c0a520d96d3ee8fac8aad/packages/start/dev/server.js#L116) (use `viteServer.resolvedUrls.local[0]`). I'm not good enough at Node (or Javascript) to know how to get the scheme from [`packages/start-node/server.js`](https://github.com/solidjs/solid-start/blob/8ef83015d5e1af8acc3c0a520d96d3ee8fac8aad/packages/start-node/server.js#L44) or [`packages/start-static/entry.js`](https://github.com/solidjs/solid-start/blob/8ef83015d5e1af8acc3c0a520d96d3ee8fac8aad/packages/start-static/entry.js#L11).